lib/http1.c

```diff
@@ -513,8 +513,11 @@ static const char *fixup_request(struct st_h2o_http1_conn_t *conn, struct phr_he
                 if (upgrade.base != NULL && h2o_contains_token(connection.base, connection.len, H2O_STRLIT("upgrade"), ',') &&
                     *entity_header_index == -1) {
                     /* early return if upgrading to h2 */
-                    if (upgrade_is_h2(upgrade) && conn->sock->ssl == NULL && conn->super.ctx->globalconf->http1.upgrade_to_http2)
-                        return fixup_request_is_h2_upgrade;
+                    if (upgrade_is_h2(upgrade) && conn->sock->ssl == NULL) {
+                        if (conn->super.ctx->globalconf->http1.upgrade_to_http2)
+                            return fixup_request_is_h2_upgrade;
+                        return NULL;
```

Comment:
I think we are on the right path, but am not sure if we can do an early return here. Notice the two additional logics below that toggles `http1_is_persistent` based on the properties of the request. I think we want to have them applied.

Reply:
PS. What do you think about 65724f0 and 20c6b77 (branch kazuho/pr3410)?